web/components/redeem-vault.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useContext, useState, useRef } from 'react';
+import { Modal } from '@components/ui/modal';
+import { tokenList } from '@components/token-swap-list';
+import { AppContext } from '@common/context';
+import { InputAmount } from './input-amount';
+import { microToReadable } from '@common/vault-utils';
+import {
+  AnchorMode,
+  contractPrincipalCV,
+  standardPrincipalCV,
+  uintCV,
+  createAssetInfo,
+  FungibleConditionCode,
+  makeStandardFungiblePostCondition,
+  noneCV,
+  someCV
+} from '@stacks/transactions';
+import { useSTXAddress } from '@common/use-stx-address';
+import { stacksNetwork as network, resolveProvider } from '@common/utils';
+import { useConnect } from '@stacks/connect-react';
+import { Alert } from './ui/alert';
+
+interface Props {
+  showRedeemModal: boolean;
+  setShowRedeemModal: (arg: boolean) => void;
+}
+
+export const RedeemVault: React.FC<Props> = ({ showRedeemModal, setShowRedeemModal, collateralToRedeem, stxVault, stStxVault, xBtcVault }) => {
+  const [state, setState] = useContext(AppContext);
+  const [errors, setErrors] = useState<string[]>([]);
+  const [redeemAmount, setRedeemAmount] = useState('');
+  const [isRedeemButtonDisabled, setIsRedeemButtonDisabled] = useState(false);
+  const stxAddress = useSTXAddress();
+  const contractAddress = process.env.REACT_APP_CONTRACT_ADDRESS || '';
+  const { doContractCall } = useConnect();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const redeemMaxAmount = () => {
+    setRedeemAmount((state.balance['usda'] / 1000000).toString());
+  };
+
+  const onInputRedeemChange = (event: any) => {
+    const value = event.target.value;
+    // trying to redeem
+    if (value > state.balance['usda'] / 1000000) {
+      if (errors.length < 1) {
+        setErrors(
+          errors.concat([`You cannot redeem more than ${state.balance['usda'] / 1000000} USDA`])
+        );
+      }
+      setIsRedeemButtonDisabled(true);
+    } else {
+      setErrors([]);
+      setIsRedeemButtonDisabled(false);
+    }
+    setRedeemAmount(value);
+  };
+
+  const redeemVaultTransact = async () => {
+    const amount = uintCV(Number((parseFloat(redeemAmount) * 1000000).toFixed(0)));
+    const postConditions = [
+      makeStandardFungiblePostCondition(
+        stxAddress || '',
+        FungibleConditionCode.Equal,
+        amount.value,
+        createAssetInfo(contractAddress, 'usda-token', 'usda')
+      ),
+    ];
+
+    const vault = collateralToRedeem === 'stSTX' ? stStxVault : collateralToRedeem === 'xBTC' ? xBtcVault : stxVault;
+    const remainingDebt = Math.max(vault.debt - redeemAmount, 0);
```

Comment:
REQ: when redeeming, the stability fee is added to the vault debt.
So this needs te become something like `(vault-debt + vault['stability-fee']) - redeemAmount`